src/utils/matching/mergeMasks.ts

```diff
@@ -0,0 +1,16 @@
+import { pathToRegExp } from 'node-match-path'
+import { Mask } from '../../setupWorker/glossary'
+
+export function mergeMasks(base: string, mask: Mask): Mask {
+  const normalizedBase = base.replace(/\/$/, '')
+
+  if (typeof mask === 'string') {
+    try {
+      return new URL(mask, base).href
+    } catch (error) {
+      return normalizedBase.concat(mask)
```

Comment:
Creating a `URL` instance may fail if the `base` string contains path parameters (not a valid URL then). Fallback to joining the strings in that case.